mythril/support/signatures.py

```diff
@@ -128,21 +113,22 @@ def write(self, path=None, sync=True):
         :return: self
         """
         path = path or self.signatures_file
-        self.signatures_file_lock = self.signatures_file_lock or SimpleFileLock(path)  # lock file to prevent concurrency issues
-        self.signatures_file_lock.aquire()  # try to aquire it within the next 10s
 
         if sync and os.path.exists(path):
             # reload and save if file exists
-            with open(path, 'r') as f:
+            with open(path, "r") as f:
+                lock_file(f)
                 sigs = json.load(f)
```

Comment:
I think we also didn't do this before but Ideally we should have the following: 
```
lock_file(f)
try:
    sigs = json.load(f)
finally:
    unlock_file(f)
```

I'm not sure how the locking libraries work in this case, but this would ensure that the lock is released.

Reply:
in unix, if the file is closed then the lock will be closed.But yes thats a better way to do it.